src/transforms/aws_cloudwatch_logs_subscription_parser.rs

```diff
@@ -0,0 +1,259 @@
+use super::Transform;
+use crate::{
+    config::{log_schema, DataType, TransformConfig, TransformContext, TransformDescription},
+    event::Event,
+    internal_events::{
+        AwsCloudwatchLogsSubscriptionParserEventProcessed,
+        AwsCloudwatchLogsSubscriptionParserFailedParse,
+    },
+};
+use chrono::{serde::ts_milliseconds, DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use std::iter;
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug, Clone, Derivative)]
+#[serde(deny_unknown_fields, default)]
+#[derivative(Default)]
+pub struct AwsCloudwatchLogsSubscriptionParserConfig {
+    pub field: Option<Atom>,
+}
+
+inventory::submit! {
+    TransformDescription::new::<AwsCloudwatchLogsSubscriptionParserConfig>("aws_cloudwatch_logs_subscription_parser")
+}
+
+#[typetag::serde(name = "aws_cloudwatch_logs_subscription_parser")]
+impl TransformConfig for AwsCloudwatchLogsSubscriptionParserConfig {
+    fn build(&self, _cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        Ok(Box::new(AwsCloudwatchLogsSubscriptionParser::from(
+            self.clone(),
+        )))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "aws_cloudwatch_logs_subscription_parser"
+    }
+}
+
+#[derive(Debug)]
+pub struct AwsCloudwatchLogsSubscriptionParser {
+    field: Atom,
+}
+
+impl From<AwsCloudwatchLogsSubscriptionParserConfig> for AwsCloudwatchLogsSubscriptionParser {
+    fn from(
+        config: AwsCloudwatchLogsSubscriptionParserConfig,
+    ) -> AwsCloudwatchLogsSubscriptionParser {
+        AwsCloudwatchLogsSubscriptionParser {
+            field: config
+                .field
+                .as_ref()
+                .unwrap_or_else(|| log_schema().message_key())
+                .clone(),
+        }
+    }
+}
+
+impl Transform for AwsCloudwatchLogsSubscriptionParser {
+    fn transform(&mut self, _event: Event) -> Option<Event> {
+        // required for trait, but transform_into is used instead
+        unimplemented!()
+    }
+
+    fn transform_into(&mut self, output: &mut Vec<Event>, event: Event) {
+        let log = event.as_log();
+
+        emit!(AwsCloudwatchLogsSubscriptionParserEventProcessed);
+
+        let message = log
+            .get(&self.field)
+            .map(|s| s.as_bytes())
+            .and_then(|to_parse| {
+                serde_json::from_slice::<AwsCloudWatchLogsSubscriptionMessage>(to_parse.as_ref())
```

Comment:
nit:

```suggestion
                serde_json::from_slice::<AwsCloudWatchLogsSubscriptionMessage>(&to_parse)
```

unless there's a reason this won't work?